### PR TITLE
Updating ScopeInfo::siUpdate comments/documentation

### PR DESCRIPTION
**Pull request by BrianBohe**
- genChangeLife is not been called any more.
- update comments which says who calls siUpdate and siUpdates header documentation

### CODE REVIEW

**Comment by BrianBohe** [comment]
@CarolEidt @sandreenko

File: src/jit/codegen.h [hunk 1]
```diff
@@ -592,7 +592,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     void siBeginBlock(BasicBlock* block);
 
     void siEndBlock(BasicBlock* block);
-
+    // Closes the "ScopeInfo" of the tracked variables that has become dead.
```
Comment (CarolEidt) [inline src/jit/codegen.h]: Here and below, this should be "have become dead."

Comment (BrianBohe) [reply src/jit/codegen.h]: Oh, thanks for that!

**Comment by BrianBohe** [comment]
Is there any way of restarting hanged processes than committing again?

**Comment by sandreenko** [comment]
@dotnet-bot test this please // It reruns Jenkins testing. 
If you need to restart Azure pipelines then it is better to ask @echesakovMSFT

**Comment by BrianBohe** [comment]
@dotnet-bot test coreclr-ci

**Comment by BrianBohe** [comment]
@dotnet-bot test coreclr-ci (Test pri0 Linux_musl x64 checked)

**Comment by echesakovMSFT** [comment]
@BrianBohe to re-run coreclr-ci you would need to go here https://github.com/dotnet/coreclr/pull/22708/checks?check_run_id=69262762, then click  [View more details on Azure Pipelines ](https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_build/results?buildId=107391) and click re-try under triple dots

**Comment by BrianBohe** [comment]
Thanks @sandreenko @echesakov !

**Comment by BrianBohe** [comment]
The checks and the details says "No problem" and "Succeeded" but this is still yellow and red. I don't know why.

**Comment by echesakovMSFT** [comment]
@BrianBohe Second attempt in https://dev.azure.com/dnceng/public/_build/results?buildId=107391 is green so you disregard what https://github.com/dotnet/coreclr/pull/22708/checks?check_run_id=69262762 says

**Comment by BrianBohe** [comment]
@echesakovMSFT Thanks!
So this is ready to merge unless there are more reviews we could attach.

**Comment by BrianBohe** [comment]
@CarolEidt @BruceForstall Is there something you would like to review?

**Pull request merged**